### PR TITLE
api: Code refactoring

### PR DESCRIPTION
**Pull request by eduardoveiga**
Signed-off-by: Eduardo Kluwe Veiga <eduardoveiga@protonmail.ch>

### CODE REVIEW

File: api/main.go [hunk 1]
```diff
@@ -109,91 +109,102 @@ const (
...
-	session, err := mgo.Dial("mongodb://mongo:27017")
-	if err != nil {
-		panic(err)
-	}
-
-	err = session.DB("main").C("devices").EnsureIndex(mgo.Index{
-		Key:        []string{"uid"},
-		Unique:     true,
-		Name:       "uid",
+func EnsureIndex(session *Session, database string, collection string, key string, unique boolean) {
```
Comment (otavio) [inline api/main.go]: As this is being done inside of main, it does not need to be public.

Comment (gustavosbarreto) [inline api/main.go]: I'm wondering if this code is actually valid for the Go compiler. Is boolean a valid data type?

Comment (gustavosbarreto) [inline api/main.go]: I'm wondering if this code is actually valid for the Go compiler. Is boolean a valid data type?

File: api/main.go [hunk 2]
```diff
@@ -109,91 +109,102 @@ const (
...
+		"active": bson.M{"$anyElementTrue": []interface{}{"$active"}}},
+
+	"match": {
+		"active": true},
+}
+
+func main() {
+	e := echo.New()
+	e.Use(middleware.Logger())
+	m := map[string]string{
```
Comment (gustavosbarreto) [inline api/main.go]: Whats does 'm' variable mean?

File: api/main.go [hunk 3]
```diff
@@ -109,91 +109,102 @@ const (
...
 
-	err = session.DB("main").C("active_sessions").EnsureIndex(mgo.Index{
-		Key:        []string{"uid"},
-		Unique:     false,
-		Name:       "uid",
-		Background: false,
-	})
-	if err != nil {
-		panic(err)
+	n := map[string]string{
```
Comment (gustavosbarreto) [inline api/main.go]: Whats does 'n' variable mean?

File: api/main.go [hunk 4]
```diff
@@ -332,24 +343,14 @@ func main() {
...
-					"localField":   "tenant_id",
-					"foreignField": "tenant_id",
-					"as":           "namespace",
-				},
+				"$lookup": user_query["lookup"],
 			},
 			{
-				"$addFields": bson.M{
-					"online":    bson.M{"$anyElementTrue": []interface{}{"$online"}},
+				"$addFields": {
```
Comment (gustavosbarreto) [inline api/main.go]: I'm sure that is not valid code because type is missing in the composite literal

**Pull request closed without merging**